### ISSUE TITLE
no-multi-spaces with VariableDeclarator = true does not work with members

### ISSUE THREAD

**Issue by QuentinRoy** [eslint/eslint#1888]
This is considered as an error:
```javascript
this.thing        = 'stuff';
this.thingyThing  = 'stuff too';
```

Actually I think it may be good just to extends key-spacing to assignment (instead of just ignoring it in no-multi-space). Thus this could be considered as a key-spacing error:
```javascript
this.badThing = 'stuff';
this.badThingyThing  = 'stuff too';
```
or of course
```javascript
var badThing = 'stuff';
var badThingyThing  = 'stuff too';
```

**Comment by QuentinRoy** [comment]
This issue may be seen as the continuation of #1524.

**Comment by nzakas** [comment]
What configuration are you using for the rule? What version of ESLint are you using? What error is output into the console?

**Comment by QuentinRoy** [comment]
```javascript
"no-multi-spaces": [1, {
    "exceptions": {
        "VariableDeclaration": true
    }
}]
```

I got "Multiple spaces found before '='. (no-multi-spaces)"
I'm using the 0.15.1 version.

**Comment by nzakas** [comment]
You need to use `VariableDeclarator`

**Comment by QuentinRoy** [comment]
My bad. That is what I did.
It perfectly works for variable declaration. Just not for members.

**Comment by nzakas** [comment]
I'm sorry, I don't understand what you're saying. What does "just not for members" mean?

**Comment by QuentinRoy** [comment]
Variables that are attached to an object.
E.g. is currently considered as an error. I think it could be a good idea to support it as well.
```javascript
this.thing        = 'stuff';
this.thingyThing  = 'stuff too';
```
This is too
```javascript
obj.thing      = 'stuff';
obj.otherThing = 'other stuff'
```

**Comment by nzakas** [comment]
Oh I see. The problem is that these are not variable declarations, they are assignment expressions. You'd need to use something like `AssignmentExpression` for this case. I'd suggest you look at http://esprima.org/demo/parse.html to figure out the correct node type to use.

**Comment by QuentinRoy** [comment]
OK I didn't got I could use any esprima type in the exception field.

However I still think it could be great if key-spacing was also dealing with assignment instead of just ignoring them in the no-multi-space rule.

I would like to see this as an error because the '=' are not aligned:
```javascript
var badThing = 'stuff';
var badThingyThing  = 'stuff too';
```

**Comment by nzakas** [comment]
Those aren't keys, so they can't be in the key spacing rule. If you want to prose a new rule, please for a new issue.
